### PR TITLE
New build system: CMake

### LINKED ISSUE #18: Use CMake
Just some notes (more or less for me):

### Swiften cmake module (I've found on github)
https://github.com/hanzz/spectrum2/blob/master/cmake_modules/SwiftenConfig.cmake

### kirigami git repo (with cmake scripts)
https://cgit.kde.org/kirigami.git/tree/README.md

This should make it easier to build apks with kirigami and to find packages like swiften. (I'm not satisfied with the current solution)

### PR DESCRIPTION
**Pull request by LNJ2**
Closes #18.

### CODE REVIEW

**Comment by LNJ2** [comment]
It's currently not possible to build an apk for android and it's not possible to use a local installation of swiften and boost, you have to install them system-wide.

But it works well.

**Comment by LNJ2** [comment]
@MRZA-MRZA We only need a local version of Swiften/Boost for SFOS/Android and for them we have to recompile the libraries for armv7.
If you want to run Kaidan on Linux you should definitely use the version from your system (`libswiften-dev` and `libboost-all-dev`)

The part that is missing is that if you want to build kaidan for SFOS/Android the libraries will be compiled for SFOS/Android and bundled with Kaidan into one package.

**Comment by MRZA-MRZA** [comment]
@LNJ2 Now it compiles but I get shared library instead of executable.
Also the project in QtCreator looks messy.

**Comment by LNJ2** [comment]
@MRZA-MRZA I haven't opened it in QtCreator.

**Comment by LNJ2** [comment]
@MRZA-MRZA have you compiled it via. cmake . ; make or with the qtcreator?

**Comment by MRZA-MRZA** [comment]
@LNJ2 I just clicked build icon in Qt Creator.

**Comment by LNJ2** [comment]
@MRZA-MRZA Then it's no wonder that it has failed. Idk what the qt creator does there, but just use `cmake . ; make -j4` as described in the readme.

**Comment by MRZA-MRZA** [comment]
@LNJ2 You won't believe but I'm getting shared library after make -j4

**Comment by LNJ2** [comment]
@MRZA-MRZA Please add some logs! I can't believe it :laughing:

**Comment by LNJ2** [comment]
https://gist.github.com/LNJ2/4129e1809158cf151654e3f80f5de131

**Comment by LNJ2** [comment]
```
lnj@ceres:~/dev/kaidanim/kaidan$ file bin/kaidan 
bin/kaidan: ELF 64-bit LSB shared object, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=ce471c5e338a2bd20a57866f07ecaf29c309fcfc, not stripped
lnj@ceres:~/dev/kaidanim/kaidan$
```

ups, I'll try to fix it :D

**Comment by LNJ2** [comment]
> This is not a bug, but due to building with -pie (Position Independent Executable), which is a hardening feature that allows address space randomization.
> 
> Some naive tools (like file) will see the resulting executable as shared library instead of executable, but they will still work as executable as normal.

Okay, so I think this is normal. (and it also was previously so)

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+find_package(Swiften REQUIRED)
+find_package(Boost REQUIRED)
+
+
+# Sources
+add_executable(${PROJECT_NAME}
+	kaidan.qrc
+	src/main.cpp
+	src/Kaidan.cpp
+	src/RosterContoller.cpp
```
Comment (MRZA-MRZA) [inline CMakeLists.txt]: https://github.com/KaidanIM/Kaidan/pull/39
Please fix it here too.

**Comment by LNJ2** [comment]
Only merge, when I have removed the WIP label. (new dependecies for the readme are missing)

**Comment by LNJ2** [comment]
@MRZA-MRZA @JBBgameich Please try out if it's building correctly now.

**Comment by JBBgameich** [comment]
I still have one error:
```
/usr/bin/ld: cannot find -lidn
collect2: error: ld returned 1 exit status
```

Probably I'm missing a dependency, but I installed everything from the readme

**Comment by LNJ2** [comment]
@JBBgameich You know that I updated it (yesterday)?
`libidn2-0-dev`, the package you are missing, is included.

**Comment by JBBgameich** [comment]
`.libidn2-0-dev ist schon die neueste Version (0.11-1).`

I've done a fresh clone from lnj2/kaidan, branch cmake, and installed all dependencies, from the readme, but there is still no change, `/usr/bin/ld: cannot find -lidn`. I had done the same before I wrote the other comment

I've done a fresh clone from lnj2/kaidan, branch cmake, and installed all dependencies, from the readme but there is still no change, `/usr/bin/ld: cannot find -lidn`. I had done the same before I wrote the other comment

**Comment by LNJ2** [comment]
@JBBgameich 
Oh, ... I think it's probably `libidn11-dev`, then.

**Comment by JBBgameich** [comment]
Yes, I know, but there was such a mess in the folder

**Comment by LNJ2** [comment]
@MRZA-MRZA Are there still any problems?

**Comment by LNJ2** [comment]
> Almost everything is good except I can run it in QtCreator:

You mean you _can't_ run it in QtCreator.

I'm not sure, but I don't think so. Why don't you just use another IDE or an editor like atom?
Can I merge it?

**Comment by JBBgameich** [comment]
I use kate for qml developenment, so I haven't tested in QtCreator too

**Pull request merged**